src/git-base.ts

```diff
@@ -31,19 +31,31 @@ export abstract class GitBaseCommand extends Command {
     this.bitBucket = await AxiosFactory.bitBucket()
   }
 
+  getJira(): AxiosJiraClient {
+    if (this.jira === undefined) {
+      this.init()
```

Comment:
Should just init itself... (AxiosFactory.jira()) but there's an issue. init and those factory calls are "infected" with async. So, you actually can't assume this.jira will be set after this line. So, it's not going to work correctly.

The heart is in the right place, but, it's bailing out to async too early. Lazily doing things is good. But... the asynchronous items in the AxiosFactory.blah()  should themselves be lazily done (inside of async functions) when required.

Aside: The factory pattern is for when you are returning something that implements an interface. This factory is moving the initialization of a type to elsewhere. I would move that initializing back into the type. This would make lazily loading it on demand within the type possible.